cli/inspector.rs

```diff
@@ -0,0 +1,204 @@
+use crate::version::DENO;
+use futures::sync::oneshot::{spawn, SpawnHandle};
+use futures::{Future, Sink, Stream};
+use serde_json::Value;
+use std::net::SocketAddrV4;
+use std::sync::mpsc::{channel, Receiver, Sender};
+use std::sync::{Arc, Mutex};
+use warp::ws::{Message, WebSocket};
+use warp::Filter;
+
+static UUID: &str = "97690037-256e-4e27-add0-915ca5421e2f";
```

Comment:
This UUID should be unguessable and randomly generated for each separate run. There is no well-defined security model for the inspector clients but unique UUID provides some guarantees that you can not just ping the inspector port and start debugging `deno` process.

In Node and Chrome there are two ways to get full web socket debugger URL:
- parse stderr of the Chrome or Node process, bot of them dumps something like Debugger is listening on ws://..., some tools use this approach.
- fetch `127.0.0.1:port/json/list` to get  json list, there is a flag to disable this endpoint in Node.

Reply:
Note that it is not just the local clients that may access localhost WebSocket server, but any web server. Just imaging malicious web site that publishes some Deno tutorial (to ensure that Deno users open the page) and then use Deno debug interface to read local FS.